gfx.c

```diff
@@ -1470,6 +1470,149 @@ void ReadNtrAnimation(char *path, struct JsonToAnimationOptions *options)
+        }
+    }
+
+    int maxNumResults = 0;
+    for (int i = 0; i < options->sequenceCount; i++)
+    {
+        for (int j = 0; j < options->sequenceData[i]->frameCount; j++)
+        {
+            int frameOffset = offset + options->sequenceData[i]->headerOffset + j * 0x8;
+            options->sequenceData[i]->frameData[j]->dataOffset = data[frameOffset] | (data[frameOffset + 1] << 8) | (data[frameOffset + 2] << 16) | (data[frameOffset + 3] << 24);
+            options->sequenceData[i]->frameData[j]->frameDelay = data[frameOffset + 4] | (data[frameOffset + 5] << 8);
+            maxNumResults++;
+            //0xBEEF
+        }
+    }
+
+    // allocate enough space for all frames to have a unique result
+    options->animationResults = malloc(sizeof(struct AnimationResults_New *) * maxNumResults);
+    for (int i = 0; i < maxNumResults; i++) 
+    {
+        options->animationResults[i] = malloc(sizeof(struct AnimationResults_New));
+        options->animationResults[i]->dataOffset = -1;
+    }
+    options->resultCount = 0;
+
+    // populate the result details
+    offset = 0x18 + (data[0x24] | (data[0x25] << 8) | (data[0x26] << 16) | (data[0x27] << 24)); //start of animation results
+    int labelOffset = offset;
+    for (int i = 0; i < options->sequenceCount; i++)
+    {
+        for (int j = 0; j < options->sequenceData[i]->frameCount; j++)
+        {
+            // check if this result ID has already been created
+            // if not, create it and store the result data
+            for (int subindex = 0; subindex < maxNumResults; subindex++) 
+            {
+                if (options->animationResults[subindex]->dataOffset == options->sequenceData[i]->frameData[j]->dataOffset)
+                {
+                    options->sequenceData[i]->frameData[j]->resultId = subindex;
+                    break;
+                }
+                if (options->animationResults[subindex]->dataOffset == -1)
+                {
+                    options->sequenceData[i]->frameData[j]->resultId = subindex;
+                    options->animationResults[subindex]->dataOffset = options->sequenceData[i]->frameData[j]->dataOffset;
+                    options->animationResults[subindex]->dataType = options->sequenceData[i]->dataType;
+                    options->animationResults[subindex]->padded = false;
+
+                    int dataOffset = offset + options->sequenceData[i]->frameData[j]->dataOffset;
+                    switch (options->sequenceData[i]->dataType) 
+                    {
+                        case 0: //index
+                            options->animationResults[subindex]->index = data[dataOffset] | (data[dataOffset + 1] << 8);
+                            labelOffset += 0x02;
+                            if (data[dataOffset + 2] == 0xCC && data[dataOffset + 3] == 0xCC)
+                            {
+                                options->animationResults[subindex]->padded = true;
```

Comment:
wanted to make a note of this. the padding here is only inserted when moving between sequences in results to keep them 4-byte aligned. so e.g. imagine:

- sequence 1 has 8 frames, 8 unique results, all type 0, total size of 16 bytes for results
- sequence 2 has 5 frames, 5 unique results, all type 0, total size of 10 bytes for results
- sequence 3 has 4 frames, 4 unique results, all type 0, total size of 8 bytes

in this scenario the results from sequence 2 are padded with 0xCCCC to align on a 4-byte boundary. a good example of this is `party_gauge.json` from pl_batt_obj in pokeplatinum, which has results padded for 3 of its sequences but other results are 4-byte aligned and go unpadded.

i store the `padded` bool on the animation result itself because it's easier but it should also be possible to drop that and determine dynamically whether a result needs to be padded based on what sequence each result belongs to, it's just harder.